File: app/src/main/java/com/arr/angel/pertpratice/db/TopicDao.java
package com.arr.angel.pertpratice.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.arr.angel.pertpratice.model.Topic;

import java.util.List;

@Dao
interface TopicDao {
    @Query("SELECT * FROM topics ORDER BY topic_id ASC")
    LiveData<List<Topic>> loadAllTopics();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertTopic(Topic topic);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateTopic(Topic topic);

    @Delete
    void deleteTopic(Topic topic);

    @Query("SELECT * FROM topics WHERE topic_id = :id")
    LiveData<Topic> loadTopicById(int id);

}

File: app/src/main/java/com/arr/angel/pertpratice/db/TopicDao.kt
package com.arr.angel.pertpratice.db

import androidx.lifecycle.LiveData
import androidx.room.*
import com.arr.angel.pertpratice.model.Topic

@Dao
interface TopicDao {
    @Query("SELECT * FROM topics ORDER BY topic_id ASC")
    fun loadAllTopics(): LiveData<List<Topic>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTopic(topic: Topic?)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateTopic(topic: Topic?)

    @Delete
    fun deleteTopic(topic: Topic?)

    @Query("SELECT * FROM topics WHERE topic_id = :id")
    fun loadTopicById(id: Int): LiveData<Topic>
}